driver-sync/src/test/functional/com/mongodb/client/AbstractRetryableWritesTest.java

```diff
@@ -139,22 +139,18 @@ public void shouldPassAllOutcomes() {
         BsonDocument operation = definition.getDocument("operation");
         BsonDocument outcome = definition.getDocument("outcome");
 
-        BsonDocument result = new BsonDocument();
-        boolean wasException = false;
         try {
-            result = helper.getOperationResults(operation);
+            BsonDocument result = helper.getOperationResults(operation);
+            assertFalse("Expected error but instead got result: " + result.toJson(),
+                    outcome.getBoolean("error", BsonBoolean.FALSE).getValue());
+            assertEquals(outcome.getDocument("result", new BsonDocument()), result.getDocument("result", new BsonDocument()));
         } catch (RuntimeException e) {
-            wasException = true;
+            if (!outcome.getBoolean("error", BsonBoolean.FALSE).getValue()) {
```

Comment:
[just an expression of my opinion]

While I know the reason behind duplicating the code `outcome.getBoolean("error", BsonBoolean.FALSE).getValue()` instead of introducing a local (`final`) variable, I strongly disagree that it is beneficial to the code in this particular case (arguing  position about this approach in general is harder, so I am not doing it now). I had to read and understand what this line of code does before I realized that it is the same code as on the line above. In other words, I as a reader did the same work twice.

Reply:
I think part of the problem here is that the expression is not so easy to parse.  You have the `BsonBoolean.FALSE` default value, and then also the `getValue()` call tacked on at the end.  So maybe the issue is more with the API that we're calling.  But regardless I introduced a local variable as you suggest.

Reply:
Thanks!

> problem here is that the expression is not so easy to parse

Absolutely. If it were `isErrorExpected(outcome)` instead of `outcome.getBoolean("error", BsonBoolean.FALSE).getValue()`, the code would be easily readable.